packages/db/src/models/base.ts

```diff
@@ -38,21 +62,59 @@ class BasePgModel<ItemType, RecordType extends { cumulus_id: number }> {
     return record.cumulus_id;
   }
 
+  /**
+   * Checks if an item is present in Postgres
+   *
+   * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+   * @param {Partial<RecordType>} params - An object or any portion of an object of type RecordType
+   * @returns {Promise<boolean>} True if the item exists, false otherwise
+   */
   async exists(
     knexOrTransaction: Knex | Knex.Transaction,
     params: Partial<RecordType>
   ): Promise<boolean> {
-    return isRecordDefined(await this.get(knexOrTransaction, params));
+    try {
+      await this.get(knexOrTransaction, params);
+      return true;
+    } catch (error) {
+      if (error instanceof RecordDoesNotExist) {
+        return false;
+      }
+      throw error;
+    }
   }
 
+  /**
+   * Creates an item in Postgres
+   *
+   * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+   * @param {ItemType} item - A record to insert into the DB
+   * @returns {Promise<number>} The ID of the inserted record
+   */
   create(
     knexOrTransaction: Knex | Knex.Transaction,
     item: ItemType
-  ) {
+  ): Promise<number> {
```

Comment:
I don't think is accurate. The return would be an array containing the cumulus ID, so `Promise<number[]>`

Reply:
Ah yeah, I'm surprised this worked. It was correctly returning an array...

The fix is here along with the new `search` method:
https://github.com/nasa/cumulus/commit/697ef411969ce58029abd0e2c43b6610b8419922